### PR TITLE
Periodically flush the reverse proxy

### PR DESCRIPTION
**Pull request by bprashanth**
The reverse proxy will keep writing to a buffered writer waiting for the connection to close, so we need to flush it for long polls.  There's a prettier way to do this, but I don't think it's worth the indirection. 

/ref https://github.com/GoogleCloudPlatform/kubernetes/issues/11678 @lavalamp

### CODE REVIEW

**Comment by lavalamp** [comment]
Is this what's breaking watch through kubectl proxy?

**Comment by bprashanth** [comment]
Yeah, well watch isn't "broken", it just won't deliver the watch notifications till enough have built up

**Comment by lavalamp** [comment]
It's odd. Is the proxy not dealing with the HTTP chunked encoding? I would expect the proxy to emit a chunk for every chunk it receives.

**Comment by bprashanth** [comment]
Don't think that's the case though there might be something more subtle going on. We flush content in the watch handler (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/apiserver/watch.go#L154), and I can write my own roundtripper and read the response body just fine, which leads me to believe that the proxy inbetween is just not flushing. Either that or it doesn't realize a chunk was emitted and ends up default flushing when the content is larger than available buffer space (because the watch will deliver notifications if one creates a bunch of services).

File: pkg/kubectl/proxy_server.go [hunk 1]
```diff
@@ -36,6 +37,14 @@ const (
 	DefaultMethodRejectRE = "POST,PUT,PATCH"
 )
 
+var (
+	// The reverse proxy will periodically flush the io writer at this frequency.
+	// Only matters for long poll connections like the one used to watch. With an
+	// interval of 0 the reverse proxy will buffer content sent on any connection
+	// with transfer-encoding=chunked.
```
Comment (lavalamp) [inline pkg/kubectl/proxy_server.go]: Add a TODO: about looking into the proxy to see if we can fix this without always adding 100ms of latency.

**Comment by lavalamp** [comment]
LGTM with a TODO

**Comment by lavalamp** [comment]
LGTM, thanks!

**Comment by stefwalter** [comment]
A fix for this had been sitting in the merge queue: #10678

**Pull request merged**